File: back/HiTribe/app/controllers/users_controller.rb
class UsersController < ApplicationController

  def login
    @user = User.find_by(username:params[:username])
    if @user
      render json: @user
    else

    end
  end

  def show
    @messages = group.messages
    render json: @messages
  end

  def friends

  end

  def groups
    @groups = User.find(params[:id]).groups
    render json: @groups
  end

  def group_users

  end

  def create
    @user = User.create(user_params)
  end

  private

  def user_params
    params.require(:user).permit(:username, :first_name, :last_name)
  end
end

File: front/HiTribe/controllers/usersController.js
function getFriends(){
  fetch(`http://localhost:3000/users/${store.currentUser}/friends`).then(function(response){
    return response.json()
  }).then(function(data){
    store.friends = []
    data.forEach(function(friend){
      new User(friend.id, friend.username, friend.first_name, friend.last_name, friend.image_url)
    })}
  )
}

function bindShowFriends(){
  $("#friends-icon").on('click', function(){
    fetch(`http://localhost:3000/groups/${store.currentGroup}/users`).then(function(response){
      return response.json()
    }).then(function(data){
      $('#group-members ul').empty()
      data.forEach(function(user){
          $('#group-members ul').append(`<li>${user.username}</li>`)
      })
    })
  })
}

function bindAddFriend(){
  $('body').on('click', '#create-friend', function(){
    let friendName = $('#add-friend-name').val()
    if(friendName){
      $.ajax({
        url: `http://localhost:3000/users/${store.currentUser}/friends`,
        method: "POST",
        dataType: 'json',
        contentType: 'application/json',
        data: JSON.stringify({friend_name: friendName, currentUser: store.currentUser})
      }).then(function(){
        getFriends()
      })

    }
  })
}
